sklearn/impute/tests/test_impute.py

```diff
@@ -1364,3 +1364,27 @@ def test_imputation_order(order, idx_order):
                                random_state=0).fit(X)
         idx = [x.feat_idx for x in trs.imputation_sequence_]
         assert idx == idx_order
+
+
+@pytest.mark.parametrize(
+    "Imputer,arg",
+    [(SimpleImputer, 'strategy'), (IterativeImputer, 'initial_strategy')])
+@pytest.mark.parametrize(
+    "strategy",
+    ['mean', 'median', 'most_frequent', "constant"])
+def test_imputation_keep_missing_features(Imputer, arg, strategy):
+    X = np.array([[1, np.nan, 2], [3, np.nan, np.nan]])
+
+    imputer = Imputer(**{arg: strategy, 'keep_missing_features': True})
+    assert X.shape == imputer.fit_transform(X).shape
+
```

Comment:
Please also check the values: that missing valued column is in the right place, and that all other values are identical to the `X_trans` produced with `keep_missing_features=False`

Reply:
This comment also applies to the common test